test/functional/ui/highlight_spec.lua

```diff
@@ -16,6 +17,91 @@ describe('color scheme compatibility', function()
   end)
 end)
 
+describe('buffer switching leaves manual syntax highlights ok', function()
+  -- When using manual syntax highlighting, it should be preserved even when
+  -- switching buffers... bug did only occur without :set hidden
+  -- Ref: vim patch 7.4.1236
+  local screen
+
+  before_each(function()
+    clear()
+    screen = Screen.new()
+    screen:attach()
+    --ignore highligting of ~-lines
+    screen:set_default_attr_ignore( {{bold=true, foreground=Screen.colors.Blue}} )
+    --syntax highlight for vimcscripts "echo"
+    screen:set_default_attr_ids( {{bold=true, foreground=Screen.colors.Brown}} )
```

Comment:
please use `[1] =`, even if it's technically redundant, it maintains the visual connection with the usages `{1:...}` below.